completions/_git-forgit

```diff
@@ -99,3 +97,30 @@ _git-forgit() {
 		;;
 	esac
 }
+
+# Check if forgit plugin is loaded
+if (( $+functions[forgit::add] )); then
+	# We're reusing existing completion functions, so load those first
+	# if not already loaded and check if completion function exists afterwards.
+	(( $+functions[_git-add] )) || _git
+	(( $+functions[_git-add] )) || return 1
+	(( $+functions[_git-branches] )) || _git-forgit
+	(( $+functions[_git-branches] )) || return 1
```

Comment:
We're loading the completion functions of git here. This is necessary because we rely on gits completion functions being present. This might slow down shell startup. Previously this was optional, because the user had to explicitly source completions/git-forgit.zsh. I think we should move this part into `_git-forgit()`. This way the first completion of forgit is slowed down, but not the shell startup, which is better in my opinion.

Reply:
So the way this seems to work is that this entire file is called when the completion is triggered the very first time.

The way I tested this.

1. put some jibberish in this file, that would fail if it gets hit.

i.e.:

```zsh
# Check if forgit plugin is loaded
if (( $+functions[forgit::add] )); then
        
       jalksdjflaksjfl;asdkjf;lakdsfj # should crash!!
      
	# We're reusing existing completion functions, so load those first
	# if not already loaded and check if completion function exists afterwards.
	(( $+functions[_git-add] )) || _git
	(( $+functions[_git-add] )) || return 1

```

I then restarted my shell, after clearing the cache.

```
rm ~/.zcomp*
zsh
```

There is no crash, indicating this file has not been sourced or read.

However, on first completion:
```zsh
> forgit::add <TAB> _git-forgit:104: command not found: jalksdjflaksjfl
_git-forgit:104: command not found: asdkjf
_git-forgit:104: command not found: lakdsfj
```

Any additional call defers to the cache, and does not hit that line again:
```zsh
... continuing session from above

Nothing to add.
> forgit::add <TAB>
README.md     completions/
```

This leads me to believe that these lines existing here are OK. This file seems to be sourced once, and only once the first time any of the completion commands in the very top line are asked for.


I would propose keeping this here. What do you think?